mwdb/resources/relations.py

```diff
@@ -111,13 +112,20 @@ def put(self, type, parent, child):
         if child_object is None:
             raise NotFound("Child object not found")
 
-        child_object.add_parent(parent_object, commit=False)
+        is_added = child_object.add_parent(parent_object, commit=False)
 
         db.session.commit()
-        logger.info(
-            "Child added",
-            extra={"parent": parent_object.dhash, "child": child_object.dhash},
-        )
+        if is_added:
+            hooks.on_created_relation(parent_object, child_object)
+            if parent_object is not child_object:
```

Comment:
Hmm `is not` seems risky here. Is it guaranteed that they will be the same SQLAlchemy object instance?

Probably yes because SQLAlchemy caches these instances somewhere and they're bound by primary key value, but check below looks safer:
```suggestion
            if parent_object.id != child_object.id:
```

Reply:
I checked it and it worked. But you are right, it looks safer. I also have applied this change in second place in code.